common/SmokeTests/SmokeTest/SmokeTest.csproj

```diff
@@ -5,26 +5,23 @@
     <LangVersion>latest</LangVersion>
     <StartupObject>SmokeTest.Program</StartupObject>
   </PropertyGroup>
-
   <ItemGroup Condition="'$(TargetFramework)' == 'net461'">
     <Reference Include="System" />
     <Reference Include="Microsoft.CSharp" />
   </ItemGroup>
-
   <ItemGroup>
     <PackageReference Include="Azure.Core" Version="1.4.0-dev.*" />
     <PackageReference Include="Azure.Identity" Version="1.2.0-dev.*" />
     <PackageReference Include="Azure.Messaging.EventHubs" Version="5.1.0-dev.*" />
```

Comment:
It probably makes sense to advance this to `5.3.0-beta` for the latest beta family or `5.2.0` for the latest GA.    Apologies; I should have remembered to do that earlier.

Reply:
These all end up getting overwritten anyway at test runtime. I wonder if it makes more sense to make them invalid placeholders, unless devs run the smoke tests manually on local without the version update polling?

Reply:
It was helpful to be able to run using the normal IDE and CLI parameters when doing the development work.   If we run the version update, I believe that would modify the `.csproj` directly and force us to remember to revert, no?    I can see that leading to me forgetting and inadvertently committing the modified version.     Though, if we look at these as "the last snapshot for when they were run that will change".... maybe that's ok?

Reply:
I think the script was written with the idea that the csproj would get overwritten in a pipelines agent context, so the commit risk wasn't there. But I think it's worth optimizing for the dev loop regardless. Perhaps we have two .csproj files, one of which is generated from the other and added to `.gitignore`? And we can have a fallback to the original so people not running the script can still run the tests.

Reply:
That's an interesting approach, and I think it covers both bases.

Reply:
I'm not sure I fully understand why we need to csproj files. It would seem we should have one committed that we thing devs can generally use and then our automation can update that one as needed to point to different versions. I believe that is the current approach we are using, although some of these resting versions need to be updated.

Reply:
After chatting with @benbp I misunderstood his approach. I think only committing a project template file with dummy placeholder values would work. We wouldn't commit the csproj so folks wouldn't be tempted to try and build an run that project they would need to do another set of sets to get the dependencies setup correctly before they could run locally. I can get behind this approach and it does align with the plans to start aggregating the smoke-tests from samples across service directories. I fully expect we will not have anything more then a template root project when we start doing that aggregation.

Reply:
@weshaggard I'm going to do this as a follow-on. Trying to keep this changeset from growing too much.